### PR TITLE
add lower bound sdk contraint analyzer tests

### PR DESCRIPTION
**Pull request by jakemac53**
This should fail for build_runner_core, will follow up with a fix for that here soon (probably just remove the set literals first)

### CODE REVIEW

File: build_config/mono_pkg.yaml [hunk 1]
```diff
@@ -6,6 +6,9 @@ stages:
     - group:
         - dartfmt: sdk
         - dartanalyzer: --fatal-infos --fatal-warnings .
+    - dartanalyzer: --fatal-infos --fatal-warnings .
+      dart:
+        - 2.0.0-dev.54
```
Comment (kevmoo) [inline build_config/mono_pkg.yaml]: Just bump this and the pubspec to 2.0.0

Comment (jakemac53) [reply build_config/mono_pkg.yaml]: We can do that separately, I want to have this tested at what is actually published

Comment (kevmoo) [reply build_config/mono_pkg.yaml]: ack

Comment (kevmoo) [inline build_config/mono_pkg.yaml]: use `2.0.0-dev.54.0` 😄

**Pull request merged**